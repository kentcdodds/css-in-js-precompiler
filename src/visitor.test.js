import path from 'path'
import stripIndent from 'strip-indent'
import * as recast from 'recast'
import * as babel from 'babel-core'
import glamorousStatic from './visitor'

test('converts static objects to css class names', () => {
  const source = `
    glamorous(
      {
        fontSize: 20,
        fontWeight: 'normal',
      },
      {
        margin: 20,
        ':hover': {
          margin: 10,
        },
      },
    )
  `
  const code = transpile(source)
  expect(code).toMatchSnapshot()
})

function transpile(source) {
  const {code} = babel.transform(stripIndent(source).trim(), {
    parserOpts: {parser: recast.parse},
    generatorOpts: {generator: recast.print, lineTerminator: '\n'},
    babelrc: false,
    plugins: [
      [glamorousStatic, {outputFile: path.join(__dirname, './out/glamor.css')}],
    ],
  })
  return code
}
