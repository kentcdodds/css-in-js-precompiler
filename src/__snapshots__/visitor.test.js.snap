// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts static objects to css class names 1`] = `
"glamorous(
  \\"css-1i7gko5\\",
  \\"css-oh2yv\\",
)"
`;
